File: app/src/main/java/com/paradoxo/amadeus/adapter/AdapterMaisDeUmItem.java
package com.paradoxo.amadeus.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.paradoxo.amadeus.R;
import com.paradoxo.amadeus.modelo.Sentenca;

import java.util.List;

public class AdapterMaisDeUmItem extends RecyclerView.Adapter<AdapterMaisDeUmItem.ViewHolder> {
    List<Sentenca> itens;

    final int VIEW_ITEM_2 = 2;
    final int VIEW_ITEM_3 = 3;
    final int VIEW_ITEM_5 = 5;

    class ViewHolder extends RecyclerView.ViewHolder {
        private TextView conteudoTexView, tituloTexView;
        private View layoutItemComum;

        private ViewHolder(View view) {
            super(view);
            tituloTexView = view.findViewById(R.id.tituloTexView);
            conteudoTexView = view.findViewById(R.id.conteudoTexView);
            layoutItemComum = view.findViewById(R.id.layoutItemComum);
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_lista_tipo_1, parent, false);
        switch (viewType) {
            case (VIEW_ITEM_2): {
                view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_lista_tipo_2, parent, false);
                break;
            }
            case (VIEW_ITEM_3): {
                view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_lista_tipo_3, parent, false);
                break;
            }

            case (VIEW_ITEM_5): {
                view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_lista_tipo_5, parent, false);
                break;
            }
        }

        return new ViewHolder(view);
    }

    @Override
    public int getItemViewType(int position) {
        int tipoView = 1;

        switch (itens.get(position).getTipo_item()) {
            case 2: {
                tipoView = VIEW_ITEM_2;
                break;
            }

            case 3: {
                tipoView = VIEW_ITEM_3;
                break;
            }

            case 5: {
                tipoView = VIEW_ITEM_5;
                break;
            }
        }

        return tipoView;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, final int position) {
        final Sentenca sentenca = itens.get(position);
        int itemViewType = getItemViewType(position);

        if (itemViewType == VIEW_ITEM_5) {
            return;
        }

        if (sentenca.getTipo_item() >= 0) {
            holder.conteudoTexView.setText(sentenca.getRespostas().get(0));

            if (itemViewType == VIEW_ITEM_3 && sentenca.getRespostas() != null) {
                holder.tituloTexView.setText(sentenca.getRespostas().get(1));
                holder.conteudoTexView.setText(sentenca.getRespostas().get(0));
            }

            holder.layoutItemComum.setOnClickListener(v -> onItemClickListener.onItemClick(v, sentenca, position));

            holder.layoutItemComum.setOnLongClickListener(v -> {
                onLongClickListener.onLongClickListener(v, position, sentenca);
                return true;
            });
        }
    }

    @Override
    public int getItemCount() {
        return itens.size();
    }

    /* Mainuplução dos eventos de click simples e longo */
    private OnItemClickListener onItemClickListener;
    private OnLongClickListener onLongClickListener;

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    public void setOnLongClickListener(OnLongClickListener onLongClickListener) {
        this.onLongClickListener = onLongClickListener;
    }

    public interface OnItemClickListener {
        void onItemClick(View view, Sentenca sentenca, int pos);
    }

    public interface OnLongClickListener {
        void onLongClickListener(View view, int position, Sentenca mensagem);
    }


    /* Manipulação da lista de dados */
    public void add(Sentenca sentenca) {
        itens.add(sentenca);
        notifyItemInserted(getItemCount());
        // notifyItemInserted(itens.size());
    }

    public void remove(int position) {
        itens.remove(position);
        notifyItemRemoved(position);
    }

    public void add(Sentenca sentenca, int posi) {
        this.itens.add(posi, sentenca);
        notifyItemChanged(posi, sentenca);
    }

    public void addAll(List<Sentenca> itens) {
        this.itens.addAll(itens);
        notifyDataSetChanged();
    }

    public List<Sentenca> getItens() {
        return itens;
    }

    public AdapterMaisDeUmItem(List<Sentenca> itens) {
        this.itens = itens;
    }

}

File: app/src/main/java/com/paradoxo/amadeus/adapter/AdapterMaisDeUmItem.kt
package com.paradoxo.amadeus.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.paradoxo.amadeus.R
import com.paradoxo.amadeus.modelo.Sentenca

class AdapterMaisDeUmItem(val itens: MutableList<Sentenca>) : RecyclerView.Adapter<AdapterMaisDeUmItem.ViewHolder>() {
    private var onItemClickListener: OnItemClickListener? = null
    private var onLongClickListener: OnLongClickListener? = null

    private val VIEW_ITEM_2 = 2
    private val VIEW_ITEM_3 = 3
    private val VIEW_ITEM_5 = 5

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val conteudoTexView: TextView? = view.findViewById(R.id.conteudoTexView)
        val tituloTexView: TextView? = view.findViewById(R.id.tituloTexView)
        val layoutItemComum: View? = view.findViewById(R.id.layoutItemComum)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutId = when (viewType) {
            VIEW_ITEM_2 -> R.layout.item_lista_tipo_2
            VIEW_ITEM_3 -> R.layout.item_lista_tipo_3
            VIEW_ITEM_5 -> R.layout.item_lista_tipo_5
            else -> R.layout.item_lista_tipo_1
        }
        return ViewHolder(LayoutInflater.from(parent.context).inflate(layoutId, parent, false))
    }

    override fun getItemViewType(position: Int): Int = when (itens[position].tipo_item) {
        2 -> VIEW_ITEM_2
        3 -> VIEW_ITEM_3
        5 -> VIEW_ITEM_5
        else -> 1
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val sentenca = itens[position]
        val itemViewType = getItemViewType(position)

        if (itemViewType == VIEW_ITEM_5) return

        if (sentenca.tipo_item >= 0) {
            holder.conteudoTexView?.text = sentenca.respostas[0]
            if (itemViewType == VIEW_ITEM_3 && sentenca.respostas.size > 1) {
                holder.tituloTexView?.text = sentenca.respostas[1]
                holder.conteudoTexView?.text = sentenca.respostas[0]
            }
            holder.layoutItemComum?.setOnClickListener { v -> onItemClickListener?.onItemClick(v, sentenca, position) }
            holder.layoutItemComum?.setOnLongClickListener { v ->
                onLongClickListener?.onLongClickListener(v, position, sentenca)
                true
            }
        }
    }

    override fun getItemCount(): Int = itens.size

    fun setOnItemClickListener(listener: OnItemClickListener) { onItemClickListener = listener }
    fun setOnLongClickListener(listener: OnLongClickListener) { onLongClickListener = listener }

    fun add(sentenca: Sentenca) { itens.add(sentenca); notifyItemInserted(itemCount) }
    fun remove(position: Int) { itens.removeAt(position); notifyItemRemoved(position) }
    fun add(sentenca: Sentenca, posi: Int) { itens.add(posi, sentenca); notifyItemChanged(posi, sentenca) }
    fun addAll(novos: List<Sentenca>) { itens.addAll(novos); notifyDataSetChanged() }
    fun interface OnItemClickListener {
        fun onItemClick(view: View, sentenca: Sentenca, pos: Int)
    }

    fun interface OnLongClickListener {
        fun onLongClickListener(view: View, position: Int, mensagem: Sentenca)
    }
}
